src/test/java/seedu/task/logic/EditCommandTest.java

```diff
@@ -6,32 +6,24 @@
 import seedu.task.logic.commands.EditTaskCommand;
 import seedu.task.model.TaskBook;
 import seedu.task.model.item.Task;
+import seedu.taskcommons.core.Messages;
 
 public class EditCommandTest extends CommandTest {
 
-    /* Need help with this test to test for duplicates not allowed. By running the program, the exception can be caught
-     * */
-
-    @Ignore
     @Test
     public void execute_editFloatTask_duplicate() throws Exception {
         // setup expectations
         TestDataHelper helper = new TestDataHelper();
         Task toBeAdded = helper.computingFloatTask();
         TaskBook expectedAB = new TaskBook();
         expectedAB.addTask(toBeAdded);
-        toBeAdded = helper.computingEditedNameFloatTask();
-        expectedAB.addTask(toBeAdded);
-        
-        // setup model
-        model.addTask(helper.computingFloatTask());
-        model.addTask(helper.computingEditedNameFloatTask());
+        Task toBeAdded2 = helper.computingEditedNameFloatTask();
+        expectedAB.addTask(toBeAdded2);
         Task toBeEdited = helper.computingFloatTask();
-        model.editTask(toBeEdited, 0);
 
         // execute command and verify result
-        assertEditTaskCommandBehavior(helper.generateAddTaskCommand(toBeAdded),helper.generateListTaskCommand(),
-                helper.generateEditTaskCommand(toBeEdited),
+        assertEditTaskCommandBehavior(helper.generateAddFloatTaskCommand(toBeAdded), helper.generateAddFloatTaskCommand(toBeAdded2),helper.generateListTaskCommand(),
+                helper.generateEditTaskCommand(toBeEdited,1),
                 String.format(EditTaskCommand.MESSAGE_DUPLICATE_TASK, toBeEdited),
                 expectedAB,
                 expectedAB.getTaskList());
```

Comment:
I'm pretty sure I did whatever you suggested...
1) Create 2 instances of toBeAdded and toBeAdded2: 

- Lines 17 - 22 added them to expectedAB
- In CommandTest.java, lines 240-242 to add them to model.

2) to call for expectedAB.editTask(toBeEdited, targetTask)

- There is no need for this step. The purpose of this test is not to have a successful adding of float task. It is to ensure that an edited task does not duplicate with an existing task in the taskbook. expectedAB is supposed to provide an expected list which should remain as the 2 original tasks being added and none edited.